index.js

```diff
@@ -153,8 +168,16 @@ NYC.prototype._wrapRequire = function () {
 
     _this.sourceMapCache.add(filename, code)
 
-    // now instrument the compiled code.
-    return _this.instrumenter.instrumentSync(code, './' + relFile)
+    var hash = md5(code)
+    var cacheFilePath = path.join(_this._cacheDirectory(), hash + '.js')
+
+    try {
+      return fs.readFileSync(cacheFilePath, 'utf8')
+    } catch (e) {
+      var instrumented = _this.instrumenter().instrumentSync(code, './' + relFile)
+      fs.writeFile(cacheFilePath, instrumented)
```

Comment:
> I suppose it's a cache so you don't need to block the process

That was my thought, but honestly [I wasn't sure](https://github.com/jamestalmage/nyc/commit/ed1a9eeb235fefc1cd21c919af3b73e29f2414ac).

> maybe add a comment to that effect?

Sounds like a plan. Is there any danger to writing async though? Should I change it back? I do not know enough about how file system failures propogate. Also, as stated [here](https://gist.github.com/jamestalmage/758b09c42069ad96c792), it doesn't seem cache collisions are happening at all in the tests I've run, so I'm not even sure what behavior it will have. Definitely needs some more testing.